### DISCUSSION TITLE
Is `fyne.NewSize()` really platform independent?

### DISCUSSION THREAD

**Discussion by eamonburns** [fyne-io/fyne#4746] [Q&A] [answered]
I have heard a lot that Fyne's sizing and positioning is platform independent. This is pretty cool, except for the fact that when I resize my window using `w.Resize(fyne.NewSize(2560, 1600)` (my monitor's dimensions), the window is exactly the same width, and only a little bit larger than my monitor. Is it just a coincidence, or is it using pixels as the units?

**Comment by andydotxyz** [comment, marked as answer]
Yes sizing is platform independent. You will see the "coincidence" of matching the pixel count if you have a low-DPI output screen on a desktop/laptop device.

**Reply by eamonburns** [reply]
Whoops! I input `1280, 800` when I meant to say `2560, 1600`. (fixed now)

Ok, so what are the actual dimensions of a full screen? Because `1600` is a little bit taller than my screen's height

**Reply by andydotxyz** [reply]
A screens dimensions will clearly depend on the screen you are using, and the users preferred scaling too.

If you want a full screen window call `Window.SetFullScreen(true)`

**Reply by eamonburns** [reply]
Wait, how can the sizing depend on the screen I am using, if it is platform independent? I thought "platform independent" meant that there would be 1 set of dimensions that would work on all platforms?

**Reply by eamonburns** [reply]
Yeah. I just did a test with multiple monitors. I set the dimensions to 1280, 800 (half the width and height of my larger monitor), and it is half the width and height of my monitor. Then I moved the window over to my smaller monitor, and it is bigger in comparison. Clearly I don't know what "platform independent" means in this context. Unless it means "no matter the OS you are using, the scaling will always use pixels"

**Reply by andydotxyz** [reply]
"platform independent" means it has no relation to pixels.
It Is difficult to describe universally because each platform has different semantics. The general concept is that irrespective of the display or pixel density the font size and actual size of the window will be consistent.

The easiest way to look at it is to have a Low DPI and a High DPI monitor (like a macOS retina) and you'll see that windows will be the same size when moved around instead of becoming half the size when you move to high DPI. If it matched pixel sizes then it would shrink.

The other way to look at it is that the window should always just do the right thing and you never have to worry about the specifics.

**Reply by eamonburns** [reply]
Ok, that is weird. When I move the window from a Low DPI monitor, to a High DPI monitor, it _does_ shrink

**Reply by eamonburns** [reply]
Could it be because I am using WSL?

**Reply by andydotxyz** [reply]
I can't comment on your computer setup. But I would recommend building windows GUI apps for Windows instead of Linux (WSL) ones.
Then you will be closer to the OS configuration instead of working through compatibility layers.

You can run fyne_demo -> Advanced to find out what the detected settings are.
